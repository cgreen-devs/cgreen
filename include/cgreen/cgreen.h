#include <stdlib.h>
#include <cgreen/unit.h>
#include <cgreen/suite.h>
#include <cgreen/text_reporter.h>
#include <cgreen/cdash_reporter.h>
#include <cgreen/cute_reporter.h>
#include <cgreen/assertions.h>
#include <cgreen/constraint_syntax_helpers.h>
#include <cgreen/runner.h>
#include <cgreen/boxed_double.h>
