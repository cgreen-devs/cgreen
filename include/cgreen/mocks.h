#ifndef MOCKS_H
#define MOCKS_H

#include <cgreen/internal/mocks_internal.h>
#include <cgreen/internal/stringify_token.h>
#include <cgreen/constraint.h>
#include <cgreen/reporter.h>

#ifdef __cplusplus
namespace cgreen {
    extern "C" {
#endif

/* Program the mocked functions to expect (or not) calls when parameters match constraints...

   expect(<function>, when(<parameter>, <constraint>), ...

   expect(<function>, when(<parameter>, <constraint>), will_return(<value>));
*/
#define expect(f, ...) expect_(get_test_reporter(), STRINGIFY_TOKEN(f), __FILE__, __LINE__, \
                               (Constraint *)__VA_ARGS__ +0, (Constraint *)0)
#define always_expect(f, ...) always_expect_(get_test_reporter(), STRINGIFY_TOKEN(f), __FILE__, __LINE__, \
                                             (Constraint *)__VA_ARGS__ +0, (Constraint *)0)
#define never_expect(f, ...) never_expect_(get_test_reporter(), STRINGIFY_TOKEN(f), __FILE__, __LINE__, \
                                           (Constraint *)__VA_ARGS__ +0, (Constraint *)0)


#ifdef _MSC_VER
// another workaround for fundamental variadic macro deficiencies in Visual C++ 2012
#define mock(...) PP_NARG(__VA_ARGS__)(get_test_reporter(), __func__, __FILE__, __LINE__, #__VA_ARGS__ "", \
                                       __VA_ARGS__)
#else
#define mock(...) PP_NARG(__VA_ARGS__)(get_test_reporter(), __func__, __FILE__, __LINE__, #__VA_ARGS__ "", \
                                       __VA_ARGS__ +0)
#endif

#define when(parameter, constraint) when_(#parameter, constraint)

#define times(number_times_called) times_(number_times_called)

/* Make Cgreen mocks strict, loose or learning */
typedef enum { strict_mocks = 0, loose_mocks = 1, learning_mocks = 2 } CgreenMockMode;
extern void cgreen_mocks_are(CgreenMockMode mode);

extern const int UNLIMITED_TIME_TO_LIVE;

#ifdef __cplusplus
    }
}
#endif

#endif
