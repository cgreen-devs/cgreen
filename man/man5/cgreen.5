.mso www.tmac
.TH CGREEN 5
.SH NAME
cgreen \- a framework for C and C++ unit tests and mocking
.SH DESCRIPTION
.B Cgreen
is a framework for creating and running compact and easy-to-read
unittests for C and/or C++.

Here are some of its features:

.IP \[bu] 2
Fluent API resulting in very readable tests
.IP \[bu]
Expressive and clear output using the default reporter
.IP \[bu]
Fully functional mocks, both strict and loose
.IP \[bu]
Each test runs in its own process for test suite robustness
.IP \[bu]
Automatic discovery and running of tests using dynamic library inspection
.IP \[bu]
Extensive and expressive constraints for many datatypes
.IP \[bu]
BDD-flavoured test declarations with Before and After declarations
.IP \[bu]
Extensible reporting mechanism
.IP \[bu]
Fully composable test suites
.IP \[bu]
An isolated test can be run in a single process for debugging


.SH "SEE ALSO"
cgreen-runner(1)
.PP
The full documentation for
.B cgreen-runner
and
.B the Cgreen framework
is in the
.B Cgreen
manual available at
.URL https://github.com/cgreen-devs/cgreen GitHub .
