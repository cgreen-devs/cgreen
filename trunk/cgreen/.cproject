<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<?fileVersion 5.0.0?>

<cproject storage_type_id="org.eclipse.cdt.core.XmlProjectDescriptionStorage2">
	<storageModule moduleId="org.eclipse.cdt.core.settings">
		<cconfiguration id="cdt.managedbuild.toolchain.gnu.base.1521338569">
			<storageModule externalCElementFile="cdt.managedbuild.toolchain.gnu.base.1521338569_org.eclipse.cdt.core.settings" id="cdt.managedbuild.toolchain.gnu.base.1521338569" name="Default"/>
			<storageModule externalCElementFile="cdt.managedbuild.toolchain.gnu.base.1521338569_cdtBuildSystem" version="4.0.0"/>
			<storageModule externalCElementFile="cdt.managedbuild.toolchain.gnu.base.1521338569_org.eclipse.cdt.core.externalSettings"/>
		</cconfiguration>
	</storageModule>
	<storageModule moduleId="cdtBuildSystem" version="4.0.0">
		<project id="cgreen.null.1958735211" name="cgreen"/>
	</storageModule>
	<storageModule moduleId="org.eclipse.cdt.core.LanguageSettingsProviders"/>
	<storageModule moduleId="org.eclipse.cdt.make.core.buildtargets">
		<buildTargets>
			<target name="all" path="" targetID="org.eclipse.cdt.build.MakeTargetBuilder">
				<buildCommand>make</buildCommand>
				<buildArguments/>
				<buildTarget>all</buildTarget>
				<stopOnError>true</stopOnError>
				<useDefaultCommand>false</useDefaultCommand>
				<runAllBuilders>true</runAllBuilders>
			</target>
			<target name="clean" path="" targetID="org.eclipse.cdt.build.MakeTargetBuilder">
				<buildCommand>rm</buildCommand>
				<buildArguments>-rf build</buildArguments>
				<buildTarget/>
				<stopOnError>true</stopOnError>
				<useDefaultCommand>false</useDefaultCommand>
				<runAllBuilders>true</runAllBuilders>
			</target>
		</buildTargets>
	</storageModule>
	<storageModule moduleId="scannerConfiguration">
		<autodiscovery enabled="true" problemReportingEnabled="true" selectedProfileId=""/>
		<scannerConfigBuildInfo instanceId="cdt.managedbuild.toolchain.gnu.base.1521338569;cdt.managedbuild.toolchain.gnu.base.1521338569.1174655132;cdt.managedbuild.tool.gnu.c.compiler.base.349579178;cdt.managedbuild.tool.gnu.c.compiler.input.1942191604">
			<autodiscovery enabled="false" problemReportingEnabled="true" selectedProfileId="org.eclipse.cdt.make.core.GCCStandardMakePerFileProfile"/>
		</scannerConfigBuildInfo>
		<scannerConfigBuildInfo instanceId="cdt.managedbuild.toolchain.gnu.base.1521338569;cdt.managedbuild.toolchain.gnu.base.1521338569.1174655132;cdt.managedbuild.tool.gnu.cpp.compiler.base.1611546606;cdt.managedbuild.tool.gnu.cpp.compiler.input.560078280">
			<autodiscovery enabled="true" problemReportingEnabled="true" selectedProfileId=""/>
		</scannerConfigBuildInfo>
	</storageModule>
</cproject>
