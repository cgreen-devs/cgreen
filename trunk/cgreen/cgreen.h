#include "unit.h"
#include "text_reporter.h"
#include "assertions.h"
#include <stdlib.h>
