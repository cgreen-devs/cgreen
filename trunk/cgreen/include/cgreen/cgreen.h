#include <cgreen/unit.h>
#include <cgreen/text_reporter.h>
#include <cgreen/cute_reporter.h>
#include <cgreen/cdash_reporter.h>
#include <cgreen/assertions.h>
#include <stdlib.h>
