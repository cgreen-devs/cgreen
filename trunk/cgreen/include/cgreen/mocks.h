#ifndef MOCKS_HEADER
#define MOCKS_HEADER

#ifdef __cplusplus
  extern "C" {
#endif

#include <inttypes.h>
#include <stdarg.h>
#include <cgreen/reporter.h>
#include <cgreen/constraint.h>

#define mock(...) mock_(__func__, #__VA_ARGS__, (intptr_t)__VA_ARGS__ +0)
#define expect(f, ...) expect_(#f, __FILE__, __LINE__, (Constraint *)__VA_ARGS__ +0, (Constraint *)0)
#define always_expect(f, ...) always_expect_(#f, __FILE__, __LINE__, (Constraint *)__VA_ARGS__ +0, (Constraint *)0)
#define expect_never(f) expect_never_(#f, __FILE__, __LINE__)
#define will_return(f, r) will_return_(#f, (intptr_t)r)
#define always_return(f, r) always_return_(#f, (intptr_t)r)
#define will_respond(f, r, ...) will_return_(#f, (intptr_t)r); expect_(#f, __FILE__, __LINE__, (Constraint *)__VA_ARGS__ +0, (Constraint *)0)
#define always_respond(f, r, ...) always_return_(#f, (intptr_t)r); always_expect_(#f, __FILE__, __LINE__, (Constraint *)__VA_ARGS__ +0, (Constraint *)0)

intptr_t mock_(const char *function, const char *parameters, ...);
void expect_(const char *function, const char *test_file, int test_line, ...);
void always_expect_(const char *function, const char *test_file, int test_line, ...);
void expect_never_(const char *function, const char *test_file, int test_line);
void will_return_(const char *function, intptr_t result);
void always_return_(const char *function, intptr_t result);
void clear_mocks();
void tally_mocks(TestReporter *reporter);

#ifdef __cplusplus
    }
#endif

#endif
