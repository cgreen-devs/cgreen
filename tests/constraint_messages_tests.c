/* Note: the whole point of these tests is that they should fail!
   This makes it hard to think straight and to give the variables good
   names ;-) */

#include <cgreen/cgreen.h>
#include <cgreen/mocks.h>
#include <signal.h>
#include <float.h>
#ifdef HAVE_SYS_RESOURCE_H
#include <sys/resource.h>
#endif


#ifdef __cplusplus
using namespace cgreen;
#endif


Describe(ConstraintMessage);
BeforeEach(ConstraintMessage) {}
AfterEach(ConstraintMessage) {}

Ensure(ConstraintMessage,for_is_null) {
    int i;
    int *pointer = &i;
    assert_that(pointer, is_null);
}

Ensure(ConstraintMessage,for_is_non_null) {
    int *pointer = NULL;
    assert_that(pointer, is_non_null);
}

Ensure(ConstraintMessage,for_is_equal_to) {
    int forty_five = 45, thirty_three = 33;
    assert_that(forty_five, is_equal_to(thirty_three));
}

Ensure(ConstraintMessage, for_is_equal_to_hex) {
    signed char chars[4];
    memset(chars, 0xaa, sizeof(chars));
    assert_that(chars[0], is_equal_to_hex(0xbb));
}

Ensure(ConstraintMessage, for_is_not_equal_to) {
    int should_not_be_forty_five = 45, forty_five = 45;
    assert_that(should_not_be_forty_five, is_not_equal_to(forty_five));
}

Ensure(ConstraintMessage, for_is_greater_than) {
    int forty_five = 45, thirty_three = 33;
    assert_that(thirty_three, is_greater_than(forty_five));
}

Ensure(ConstraintMessage, for_is_less_than) {
    int forty_five = 45, thirty_three = 33;
    assert_that(forty_five, is_less_than(thirty_three));
}


// Contents of struct/memory
Ensure(ConstraintMessage, for_is_equal_to_contents_of) {
    int forty_five[45] = {45, 44, 43}, thirty_three[33] = {45, 44, 33};
    assert_that(thirty_three, is_equal_to_contents_of(forty_five, 55));
}

Ensure(ConstraintMessage, for_is_not_equal_to_contents_of) {
    int forty_five_and_up[3] = {45, 46, 47}, another_forty_five_and_up[3] = {45, 46, 47};
    assert_that(forty_five_and_up, is_not_equal_to_contents_of(another_forty_five_and_up, 3));
}


//  Strings
Ensure(ConstraintMessage, for_is_equal_to_string) {
    const char *forty_five = "this string is fortyfive", *thirty_three = "this string is thirtythree";
    assert_that(thirty_three, is_equal_to_string(forty_five));
}

Ensure(ConstraintMessage, for_is_not_equal_to_string) {
    const char *forty_five = "this string is fortyfive", *another_forty_five = "this string is fortyfive";
    assert_that(another_forty_five, is_not_equal_to_string(forty_five));
}

Ensure(ConstraintMessage, for_contains_string) {
    const char *forty_five = "fortyfive", *not_containing_forty_five = "this text is thirtythree";
    assert_that(not_containing_forty_five, contains_string(forty_five));
}

Ensure(ConstraintMessage, for_does_not_contain_string) {
    const char *contains_forty_five = "this string is fortyfive", *forty_five = "fortyfive";
    assert_that(contains_forty_five, does_not_contain_string(forty_five));
}

Ensure(ConstraintMessage, for_begins_with_string) {
    const char *does_not_begin_with_forty_five = "this string does not begin with fortyfive", *forty_five = "fortyfive";
    assert_that(does_not_begin_with_forty_five, begins_with_string(forty_five));
}

Ensure(ConstraintMessage, for_does_not_begin_with_string) {
    const char *begins_with_forty_five = "fortyfive is the start of this", *forty_five = "fortyfive";
    assert_that(begins_with_forty_five, does_not_begin_with_string(forty_five));
}

Ensure(ConstraintMessage, for_ends_with_string) {
    const char *does_not_end_with_forty_five = "fortyfive is not the end of this string", *forty_five = "fortyfive";
    assert_that(does_not_end_with_forty_five, ends_with_string(forty_five));
}

Ensure(ConstraintMessage, for_does_not_end_with_string) {
    const char *ends_with_forty_five = "this string ends with fortyfive", *forty_five = "fortyfive";
    assert_that(ends_with_forty_five, does_not_end_with_string(forty_five));
}


// Double

Ensure(ConstraintMessage, for_is_equal_to_double) {
    double four_point_five = 4.5f, three_point_three = 3.3f;
    assert_that_double(four_point_five, is_equal_to_double(three_point_three));
}

Ensure(ConstraintMessage, for_is_not_equal_to_double) {
    significant_figures_for_assert_double_are(4);
    double epsilon = 1.0e-4;
    double four_point_five = 4.5, almost_four_point_five = 4.5 - epsilon;
    assert_that_double(four_point_five, is_not_equal_to_double(almost_four_point_five));
}

Ensure(ConstraintMessage, for_is_greater_than_double) {
    double four_point_five = 4.5f, three_point_three = 3.3f;
    assert_that_double(three_point_three, is_greater_than_double(four_point_five));
}

Ensure(ConstraintMessage, for_is_less_than_double) {
    double four_point_five = 4.5f, three_point_three = 3.3f;
    assert_that_double(four_point_five, is_less_than_double(three_point_three));
}

Ensure(ConstraintMessage, for_equal_to_double) {
  assert_that_double(0, is_equal_to_double(1));
}

Ensure(ConstraintMessage, for_equal_to_double_negative) {
  assert_that_double(-1, is_equal_to_double(-2));
}

Ensure(ConstraintMessage, for_not_equal_to_double) {
  assert_that_double(0, is_not_equal_to_double(0));
}

Ensure(ConstraintMessage, for_not_equal_to_double_negative) {
  assert_that_double(-1, is_not_equal_to_double(-1));
}

Ensure(ConstraintMessage, for_is_less_than_double_with_accuracy) {
  significant_figures_for_assert_double_are(4);
  assert_that_double(1.0, is_less_than_double(1.0 - 1.0e-3 - DBL_EPSILON));
}

Ensure(ConstraintMessage, for_is_greater_than_double_with_accuracy) {
  significant_figures_for_assert_double_are(4);
  assert_that_double(1.0, is_greater_than_double(1.0 + 1.0e-3 + DBL_EPSILON));
}


// Basic core assert_that()

Ensure(ConstraintMessage, for_assert_that) {
    assert_that(0 == 1);
}


// Mocks

void some_mock(int parameter) {
    mock(parameter);
}

Ensure(ConstraintMessage, for_mock_called_without_expectation) {
    some_mock(0);
}

Ensure(ConstraintMessage, for_mock_called_more_times_than_expected) {
    expect(some_mock);
    some_mock(0);
    some_mock(0);
}

Ensure(ConstraintMessage, for_mock_called_with_unexpected_parameter_value) {
    expect(some_mock, when(parameter, is_equal_to(1)));
    some_mock(0);
}

/* Tests that will fail with "illegal" type of messages */
Ensure(ConstraintMessage, for_always_followed_by_expectation) {
    always_expect(some_mock, when(parameter, is_equal_to(1)));
    expect(some_mock, when(parameter, is_equal_to(0)));
}

Ensure(ConstraintMessage, for_violated_never_expect) {
    never_expect(some_mock, when(parameter, is_equal_to(1)));
    some_mock(1);
}

Ensure(ConstraintMessage, for_mock_parameter_name_not_matching_constraint_parameter_name) {
    expect(some_mock, when(PARAMETER, is_equal_to(0)));
    some_mock(0);
}

void forgot_to_pass_parameters_mock(int x) {
    (void)x; // to silence compiler warning
    mock();
}

Ensure(ConstraintMessage, for_no_mock_parameters_with_parameter_constraint) {
    expect(forgot_to_pass_parameters_mock, when(x, is_equal_to(0)));
    forgot_to_pass_parameters_mock(0);
}

Ensure(ConstraintMessage, increments_exception_count_when_terminating_via_SIGQUIT) {
#ifdef HAVE_SYS_RESOURCE_H
    struct rlimit core_limit;
    core_limit.rlim_cur = 1U;
    core_limit.rlim_max = 1U;
    setrlimit(RLIMIT_CORE, &core_limit);
#endif
    raise(SIGQUIT);
}

Ensure(ConstraintMessage, increments_exception_count_when_terminating_via_SIGTERM) {
    raise(SIGTERM);
}

#ifdef __cplusplus
Ensure(ConstraintMessage, for_incorrect_assert_throws) {
    assert_throws(std::string, throw "something else");
}

Ensure(ConstraintMessage, for_exception_was_expected_but_nothing_thrown) {
    assert_throws(std::string, (void)5);
}
#endif
