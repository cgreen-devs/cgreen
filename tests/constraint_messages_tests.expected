Running "constraint_messages_tests" (37 tests)...
constraint_messages_tests.c:000: Failure: ConstraintMessage -> for_always_followed_by_expectation 
	Mocked function [some_mock] already has an expectation that it will always be called a certain way; any expectations declared after an always expectation are invalid

constraint_messages_tests.c:000: Failure: ConstraintMessage -> for_assert_that 
	Expected [0 == 1] to [be true]

constraint_messages_tests.c:000: Failure: ConstraintMessage -> for_begins_with_string 
	Expected [does_not_begin_with_forty_five] to [begin with string] [forty_five]
		actual value:			["this string does not begin with fortyfive"]
		expected to begin with:		["fortyfive"]

constraint_messages_tests.c:000: Failure: ConstraintMessage -> for_contains_string 
	Expected [not_containing_forty_five] to [contain string] [forty_five]
		actual value:			["this text is thirtythree"]
		expected to contain:		["fortyfive"]

constraint_messages_tests.c:000: Failure: ConstraintMessage -> for_does_not_begin_with_string 
	Expected [begins_with_forty_five] to [not begin with string] [forty_five]
		actual value:			["fortyfive is the start of this"]
		expected to not begin with:	["fortyfive"]

constraint_messages_tests.c:000: Failure: ConstraintMessage -> for_does_not_contain_string 
	Expected [contains_forty_five] to [not contain string] [forty_five]
		actual value:			["this string is fortyfive"]
		expected to not contain:	["fortyfive"]

constraint_messages_tests.c:000: Failure: ConstraintMessage -> for_does_not_end_with_string 
	Expected [ends_with_forty_five] to [not end with string] [forty_five]
		actual value:			["this string ends with fortyfive"]
		expected to not end with:	["fortyfive"]

constraint_messages_tests.c:000: Failure: ConstraintMessage -> for_ends_with_string 
	Expected [does_not_end_with_forty_five] to [end with string] [forty_five]
		actual value:			["fortyfive is not the end of this string"]
		expected to end with:		["fortyfive"]

constraint_messages_tests.c:000: Failure: ConstraintMessage -> for_equal_to_double 
	Expected [0] to [equal double] [1] within [8] significant figures
		actual value:			[0.000000]
		expected value:			[1.000000]

constraint_messages_tests.c:000: Failure: ConstraintMessage -> for_equal_to_double_negative 
	Expected [-1] to [equal double] [-2] within [8] significant figures
		actual value:			[-1.000000]
		expected value:			[-2.000000]

constraint_messages_tests.c:000: Failure: ConstraintMessage -> for_is_equal_to 
	Expected [forty_five] to [equal] [thirty_three]
		actual value:			[45]
		expected value:			[33]

constraint_messages_tests.c:000: Failure: ConstraintMessage -> for_is_equal_to_contents_of 
	Expected [thirty_three] to [equal contents of] [forty_five]
		at offset:			[2]
			actual value:		[0x21]
			expected value:		[0x2b]

constraint_messages_tests.c:000: Failure: ConstraintMessage -> for_is_equal_to_double 
	Expected [four_point_five] to [equal double] [three_point_three] within [8] significant figures
		actual value:			[4.500000]
		expected value:			[3.300000]

constraint_messages_tests.c:000: Failure: ConstraintMessage -> for_is_equal_to_hex 
	Expected [(unsigned char) bytes[0]] to [equal] [0xbb]
		actual value:			[0xaa]
		expected value:			[0xbb]

constraint_messages_tests.c:000: Failure: ConstraintMessage -> for_is_equal_to_string 
	Expected [thirty_three] to [equal string] [forty_five]
		actual value:			["this string is thirtythree"]
		expected to equal:		["this string is fortyfive"]

constraint_messages_tests.c:000: Failure: ConstraintMessage -> for_is_greater_than 
	Expected [thirty_three] to [be greater than] [forty_five]
		actual value:			[33]
		expected to be greater than:	[45]

constraint_messages_tests.c:000: Failure: ConstraintMessage -> for_is_greater_than_double 
	Expected [three_point_three] to [be greater than double] [four_point_five] within [8] significant figures
		actual value:			[3.300000]
		expected value:			[4.500000]

constraint_messages_tests.c:000: Failure: ConstraintMessage -> for_is_greater_than_double_with_accuracy 
	Expected [1.0] to [be greater than double] [1.0 + 1.0e-3 + DBL_EPSILON] within [4] significant figures
		actual value:			[1.000000]
		expected value:			[1.001000]

constraint_messages_tests.c:000: Failure: ConstraintMessage -> for_is_less_than 
	Expected [forty_five] to [be less than] [thirty_three]
		actual value:			[45]
		expected to be less than:	[33]

constraint_messages_tests.c:000: Failure: ConstraintMessage -> for_is_less_than_double 
	Expected [four_point_five] to [be less than double] [three_point_three] within [8] significant figures
		actual value:			[4.500000]
		expected value:			[3.300000]

constraint_messages_tests.c:000: Failure: ConstraintMessage -> for_is_less_than_double_with_accuracy 
	Expected [1.0] to [be less than double] [1.0 - 1.0e-3 - DBL_EPSILON] within [4] significant figures
		actual value:			[1.000000]
		expected value:			[0.999000]

constraint_messages_tests.c:000: Failure: ConstraintMessage -> for_is_non_null 
	Expected [pointer] to [be non null]

constraint_messages_tests.c:000: Failure: ConstraintMessage -> for_is_not_equal_to 
	Expected [should_not_be_forty_five] to [not equal] [forty_five]
		actual value:			[45]

constraint_messages_tests.c:000: Failure: ConstraintMessage -> for_is_not_equal_to_contents_of 
	Expected [forty_five_and_up] to [not equal contents of] [another_forty_five_and_up]

constraint_messages_tests.c:000: Failure: ConstraintMessage -> for_is_not_equal_to_double 
	Expected [four_point_five] to [not equal double] [almost_four_point_five] within [4] significant figures
		actual value:			[4.500000]
		expected value:			[4.499900]

constraint_messages_tests.c:000: Failure: ConstraintMessage -> for_is_not_equal_to_string 
	Expected [another_forty_five] to [not equal string] [forty_five]
		actual value:			["this string is fortyfive"]

constraint_messages_tests.c:000: Failure: ConstraintMessage -> for_is_null 
	Expected [pointer] to [be null]

constraint_messages_tests.c:000: Failure: ConstraintMessage -> for_mock_called_more_times_than_expected 
	Mocked function [some_mock] was called too many times

constraint_messages_tests.c:000: Failure: ConstraintMessage -> for_mock_called_with_unexpected_parameter_value 
	Expected [[parameter] parameter in [some_mock]] to [equal] [1]
		actual value:			[0]
		expected value:			[1]

constraint_messages_tests.c:000: Failure: ConstraintMessage -> for_mock_called_without_expectation 
	Mocked function [some_mock] did not have an expectation that it would be called

constraint_messages_tests.c:000: Failure: ConstraintMessage -> for_mock_parameter_name_not_matching_constraint_parameter_name 
	Mocked function [some_mock] did not define a parameter named [PARAMETER]. Did you misspell it in the expectation or forget it in the mock's argument list?

constraint_messages_tests.c:000: Failure: ConstraintMessage -> for_no_mock_parameters_with_parameter_constraint 
	Mocked function [forgot_to_pass_parameters_mock] did not define a parameter named [x]. Did you misspell it in the expectation or forget it in the mock's argument list?

constraint_messages_tests.c:000: Failure: ConstraintMessage -> for_not_equal_to_double 
	Expected [0] to [not equal double] [0] within [8] significant figures
		actual value:			[0.000000]
		expected value:			[0.000000]

constraint_messages_tests.c:000: Failure: ConstraintMessage -> for_not_equal_to_double_negative 
	Expected [-1] to [not equal double] [-1] within [8] significant figures
		actual value:			[-1.000000]
		expected value:			[-1.000000]

constraint_messages_tests.c:000: Failure: ConstraintMessage -> for_violated_never_expect 
	Mocked function [some_mock] has an expectation that it will never be called, but it was

constraint_messages_tests.c:000: Exception: ConstraintMessage -> increments_exception_count_when_terminating_via_SIGQUIT 
	Test terminated with signal: Quit

constraint_messages_tests.c:000: Exception: ConstraintMessage -> increments_exception_count_when_terminating_via_SIGTERM 
	Test terminated with signal: Terminated

  "ConstraintMessage": 35 failures, 2 exceptions in 0ms.
Completed "constraint_messages_tests": 35 failures, 2 exceptions in 0ms.
